apps/sms/views/conversation/js/conversation.js

```diff
@@ -1160,10 +1156,36 @@ var ConversationView = {
 
   /**
    * Navigates user to Composer or Thread panel with custom parameters.
-   * @param {*} parameters Optional navigation parameters.
+   * @param {Object} parameters Navigation parameters. `number` and `messageId`
+   * are mutually exclusive in the current implementation.
+   * @param {String} [parameters.number] Phone number or e-mail to send a
+   * message to.
+   * @param {Number} [parameters.messageId] Template message to resend.
    * @returns {Promise} Promise that is resolved once navigation is completed.
    */
-  navigateToComposer: function(parameters) {
+  initiateNewMessage: function(parameters) {
+    var navigateToComposer = () => {
+      var draftCreatePromise;
+
+      if (parameters.messageId) {
+        draftCreatePromise = this.storeDraftFromMessage(parameters.messageId);
+      } else if (parameters.number) {
+        var draft = new Draft({
+          recipients: [parameters.number],
+          type: Utils.isEmailAddress(parameters.number) ? 'mms' : 'sms'
+        });
+        draftCreatePromise = Drafts.add(draft).store().then(() => draft.id);
+      } else {
+        throw new TypeError('Unknown parameter');
+      }
+
+      var focusComposer = !!(parameters && parameters.number);
+      return draftCreatePromise.then(
+        (draftId) => Navigation.toPanel('composer', { draftId, focusComposer })
+      );
+    };
+
+
     var draftDiscardPromise;
```

Comment:
Looks like discard promise is not properly analyzed, if you open thread with the message that contains phone number, enter some text into message input and then will try to send message to a number from message - app will ask you if you want to discard unsent message. And it doesn't matter what you choose, your message will be discarded.

As we talked offline - this dialog will be removed soon anyway, so it's not a big deal, we just need to check if it can't cause something more serious.